src/evict/evict_lru.c

```diff
@@ -591,8 +591,12 @@ __evict_update_work(WT_SESSION_IMPL *session)
     /* Build up the new state. */
     flags = 0;
 
+    /* Debug mode is set at connection open and reconfigure, need to maintain what was set. */
+    if (F_ISSET(cache, WT_CACHE_EVICT_DEBUG_MODE))
```

Comment:
Though this change is correct. There is a discrepancy between how we track connection level debug flags in ```conn->debug_flags``` instead of ```conn->cache->flags``` where we risk overwriting. It might be better to switch the evict debug mode flag to reside in ```conn->debug_flags```. An example of such a flag is ```WT_CONN_DEBUG_UPDATE_RESTORE_EVICT``` :

```
    if (FLD_ISSET(conn->debug_flags, WT_CONN_DEBUG_UPDATE_RESTORE_EVICT)) {
        LF_SET(WT_CACHE_EVICT_SCRUB);
        LF_CLR(WT_CACHE_EVICT_NOKEEP);
    }
```

I looked at the change for WT-4477 and what might have happened is that we introduced ```conn->debug_flags``` at a later date and likely forgot to update the flag for evict_debug_mode.

Reply:
Should debug_mode.table_logging be also switched to debug_flags instead of log_flags to have all debug mode options in the one place? I think that would make it more consistent.